### PR TITLE
Comments AutoCompletion

### PR DESCRIPTION
**Pull request by blizzz**
Follow up to #6779, both backend and gui, for #2443

I also did a quick demo video [demo video](https://cloud.nextcloud.com/s/BdlSXmoHCmkJk9M)

* Type @ and a char to start auto completion for a user
* Mentiones are rendered just like as we use avatar + displayname everywhere
* works also when editing users
* user shared with and commented are sorted on top (given their are part of the result set)
* it's themed!
* known issue on FF: to remove a mention, it must be marked first. Apparently a browser bug. On Chromium, backspace works fine.

Please test and review

### CODE REVIEW

**Review by jancborchardt — APPROVED** [review]
Looks nice from the demo video! I’d say further fixes should be done in follow-ups cause this is too big already. :)

**Comment by nickvergessen** [comment]
Should only ship the minified js file, instead of the examples and other unneeded stuff.

File: lib/private/Server.php [hunk 1]
```diff
@@ -1011,6 +1011,11 @@ public function __construct($webRoot, \OC\Config $config) {
 		});
 		$this->registerAlias('CollaboratorSearch', \OCP\Collaboration\Collaborators\ISearch::class);
 
+		$this->registerService(\OCP\Collaboration\AutoComplete\IManager::class, function (Server $c) {
+			return new Collaboration\AutoComplete\Manager($c);
```
Comment (nickvergessen) [inline lib/private/Server.php]: please import completely

Comment (nickvergessen) [reply lib/private/Server.php]: actually you should be able to drop this completely because of our magic automatic injection

**Comment by nickvergessen** [comment]
1. If you match the full user id with `@test1` the user is not suggested anymore, but `@test` brings them up
2. The avatar is not rendered anymore after you clicked/selected one of the options.

**Comment by blizzz** [comment]
@nickvergessen great testing 👍 fixed with recent commits

**Review by rullzer — APPROVED** [review]
Awesome

**Comment by blizzz** [comment]
🍻

**Pull request merged**